server/src/feature/validateTextDocument.ts

```diff
@@ -0,0 +1,42 @@
+import { CharStreams, CommonTokenStream } from 'antlr4ts';
+import { Diagnostic, TextDocumentChangeEvent } from 'vscode-languageserver-protocol';
+
+import { YmlLexer, YmlParser } from '../grammar';
+import Session from '../session';
+import { YmlKaoFileVisitor, YmlParsingErrorListener } from '../visitors';
+
+export default function(_session: Session): any {
+    return (textDocumentChangeEvent: TextDocumentChangeEvent): void => {
+        const textDocument = textDocumentChangeEvent.document;
+        _session.log(`Yseop.vscode-yseopml − Parsing ${textDocument.uri}`);
+        const textDocUri = textDocument.uri;
+        const diagnostics: Diagnostic[] = [];
+
+        // Create the lexer and parser
+        const inputStream = CharStreams.fromString(textDocument.getText());
+        const lexer = new YmlLexer(inputStream);
+        const tokenStream = new CommonTokenStream(lexer);
+        const parser = new YmlParser(tokenStream);
+        parser.removeErrorListeners();
+        parser.addErrorListener(new YmlParsingErrorListener(diagnostics));
```

Comment:
Maybe naive question: do we recreate and redo _all this_ at each change of a file? :open_mouth: